sync.sh

```diff
@@ -72,7 +72,7 @@ echo "$jobs" | jq 'sort_by(.enqueue_time)[]' -c | while IFS= read -r job; do
     echo "$job" | jq . > "$tdir/job.json"
 
     # Download images
-    echo "$job" | jq '.image_paths[]' -c -r | while IFS= read -r img_url; do
+    echo "$job" | jq '.image_paths[]' -c -r | while IFS=$'\t\r' read -r img_url; do
```

Comment:
Interesting, I'm not getting any newlines in the input to curl. (GNU bash, version 5.1.4 -- Debian Linux 11/bullseye.) The `read` builtin should be splitting on lines, and only using IFS for word delimiters.

I don't think tab should be required, but if CR is what you're running into... are you perhaps running this under Cygwin, on Windows? That might explain the presence of CR characters, which I wouldn't expect to encounter on Linux or Mac (which should only be producing LF characters).

Reply:
Yes, sorry, I'm running this in bash on windows

Reply:
Ah, OK. Hmm. I guess jq must be producing CRs on Windows. I'd like to preserve the "no word delimiters" `read` call, so maybe this would be the cleanest approach:

```suggestion
    # Turn CRLF into LF (needed when jq is run under Cygwin)
    echo "$job" | jq '.image_paths[]' -c -r | sed 's/\r$//' | while IFS= read -r img_url; do
```